core/DependencyInjection/ContainerBuilder.php

```diff
@@ -227,4 +229,9 @@ private function isApplicationInstalled(): bool
     {
         return file_exists(rtrim((string)$this->configPath, '/') . '/generis/installation.conf.php');
     }
+
+    private function ensureCacheDirectoryIsCreated(): void
+    {
+        @mkdir($this->cachePath, 0700, true);
```

Comment:
Could we check if directory exist or is writable and create it otherwise, instead of using the `@` operator which will mask permission issues, etc?

I think this is something that we should solve on DevOps side, cause for config/data we break the installation if they do not exist and are not writable. 

Wdyt?

Reply:
Updated the code to check for directory beforehand.

Whether the solution should be on infrastructure or application side, at first I wanted to solve it on infra code, but I think the argument is valid that app should ensure the folder structure it expects. And it is reasonable for infra to assume that `data` folder can be wiped, any files that must persist are stored in specifically configured file-systems.

Reply:
I suggest to add this as a pre-install/pre-update script, cause DI container should not be creating generis cache:

```
if (!mkdir(GENERIS_CACHE_PATH, 0700, true) && !is_dir(GENERIS_CACHE_PATH)) {
    throw new InvalidArgumentException(
        sprintf(
            'Could not create application cache',
            $this->cachePath
        )
    );
}
```